### PR TITLE
Clarify structure types

### PR DESCRIPTION
**Pull request by tychonievich**
As discussed in #178, there are ambiuguities in the current spec regarding structure types, enumeration values, and extensions. This PR attempts to resolve those ambiguities.

### CODE REVIEW

File: specification/gedcom-02-datatypes.md [hunk 1]
```diff
@@ -47,6 +47,22 @@ and have their URI defined by the schema.
 Each enumeration value has a distinct meaning
 as identified by its corresponding URI.
 
+The URI of a given tag in an enumeration payload is determined by the tag itself and by the structure type of the structure it is in the payload of.
+
+:::example
+The tag `HUSB` is used in this document to represent two enumeration values.
+Which one is meant can be identified by the structure type is appears in as follows:
```
Comment (tychonievich) [inline specification/gedcom-02-datatypes.md:54]:
```suggestion
Which one is meant can be identified by the structure type it appears in as follows:
```

**Review by dthaler — APPROVED** [review]

File: specification/gedcom-01-serialization.md [hunk 2]
```diff
@@ -272,6 +304,10 @@ The recommended way to go beyond the set of standard structure types in this spe
 
 This specification also provides multiple ways for extension authors to go beyond the specification without submitting a feature request, which are described in the remainder of this section.
 
+Extensions can introduce new structure types, new enumeration values, new calendars with their associated months, and new datatypes.
+They can also extend existing existing structures with new permitted substructure types and existing enumeration-type payloads with new permitted values.
```
Comment (tychonievich) [inline specification/gedcom-01-serialization.md:308]:
```suggestion
They can also extend existing structures with new permitted substructure types and existing enumeration-type payloads with new permitted values.
```

Remove duplicate word

File: specification/gedcom-01-serialization.md [hunk 3]
```diff
@@ -272,6 +306,10 @@ The recommended way to go beyond the set of standard structure types in this spe
 
 This specification also provides multiple ways for extension authors to go beyond the specification without submitting a feature request, which are described in the remainder of this section.
 
+Extensions can introduce new structure types, new enumeration values, new calendars with their associated months, and new datatypes.
+They can also extend existing structures with new permitted substructure types and existing enumeration-type payloads with new permitted values.
```
Comment (tychonievich) [inline specification/gedcom-01-serialization.md:310]:
```suggestion
They can also extend existing structures with new permitted substructure types and extend existing enumeration-type payloads with new permitted values.
```

**Review by lynchrs3 — APPROVED** [review]

File: specification/gedcom-01-serialization.md [hunk 4]
```diff
@@ -151,6 +165,26 @@ The **tag** matches production `Tag` and encodes the structure's type.
 Tags that match the production `stdTag` are defined in this document.
 Tags that match `extTag` are defined according to [Extensions].
 
+The same tag may be used to represent multiple structure types.
+The structure type of each structure is identified by its tag and the type of its superstucture.
```
Comment (tychonievich) [inline specification/gedcom-01-serialization.md:169]:
```suggestion
The structure type of each structure is identified by its tag and the type of its superstructure.
```

**Pull request merged**